Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,417 @@
+                    }
+                    var icon = EditorGUIUtility.IconContent("winbtn_win_min_h");
+
+                    if (GUILayout.Button(icon, GUILayout.Width(30)))
+                    {
+                        migrationObjects.Remove(migrationObject);
+                        break;
+                    }
+                }
+            }
+            EditorGUILayout.Space();
+
+            using (new GUILayout.VerticalScope(EditorStyles.helpBox))
+            {
+                if (migrationObjects.Count == 0)
+                {
+                    GUI.enabled = false;
+                }
+
+                if (GUILayout.Button("Migrate"))
+                {
+                    MigrateSelection();
+                }
+
+                using (new EditorGUILayout.HorizontalScope())
+                {
+                    if (migrationObjects.Count > 0)
+                    {
+                        GUILayout.FlexibleSpace();
+
+                        string tooltip = $"{migrationObjects.Count} {toolbarTitles[(int)selectedToolbar]} selected for migration";
+                        EditorGUILayout.LabelField(new GUIContent(tooltip, InspectorUIUtility.WarningIcon));
+                    }
+                }
+            }
+            EditorGUILayout.EndScrollView();
+        }
+
+        private void AddSelection()
+        {
+            if (selectedToolbar == ToolbarOption.Objects)
+            {
+                if (!migrationObjects.Contains(selection))
+                {
+                    migrationObjects.Add(selection);
+                }
+            }
+            else
+            {
+                if (!migrationObjects.Contains(selection) && IsExtension(AssetDatabase.GetAssetPath(selection), SceneExtension))
+                {
+                    migrationObjects.Add(selection);
+                }
+            }
+        }
+
+        private void MigrateAllScenes()
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+                 
+            var previousScenePath = EditorSceneManager.GetActiveScene().path;
+
+            String[] scenePaths = FindAllAssets(SceneExtension);
+
+            for (int i = 0; i < scenePaths.Length; i++)
+            {
+                MigrateScene(scenePaths[i]);
+            }
+        }
+
+        private void MigrateScene(String path)
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+
+            var previousScenePath = EditorSceneManager.GetActiveScene().path;
+            Scene currentScene = EditorSceneManager.OpenScene(path);
+
+            foreach (var parent in currentScene.GetRootGameObjects())
+            {
+                MigrateGameObjectHierarchy(parent);
+            }
+            EditorSceneManager.SaveScene(currentScene);
+            EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
```

Comment:
If we're going to migrate all scenes, does it make sense to always open the previous scene with each scene migration, or should we only do that at the end?

To be clear, here's what happens today:
1) Have Scene 1 open. I click migrate all.
2) Scene 2 is opened, migrated, then Scene 1 is opened.
3) Scene 3 is opened, migrated, then Scene 1 is opened.
4) Scene 4 is opened, migrated, then Scene 1 is opened.

It feels like instead we should really just open Scene 1 at the end.

Reply:
No it doesn't. :)

Reply:
I thought this was happening too? In Will's example, does `previousScenePath` not point to _Scene 1_ while we open and migrate `currentScene`. Then wouldn't the following line of code reopen _Scene 1_?
```csharp
EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
```
This would make sense when migrating a single scene, but we call this in `MigrateAllScenes` too, so this line of code should be called for as many paths as there are in `scenePaths`. Let me know if I've misunderstood.

Reply:
Changed logic